### ISSUE TITLE
range(VALUE, STREAM)

### ISSUE THREAD

**Issue by pkoppstein** [stedolan/jq#886]
The Advanced Topics page (https://github.com/stedolan/jq/wiki/Advanced-Topics)
says: 

> range(0,1;3,4) is the same as range(0;3), range(0;3), range(1;3), range(1;4)

This was evidently intended to read:

> range(0,1;3,4) is the same as range(0;3), range(0;4), range(1;3), range(1;4)

but in any case, jq 1.4 and jq 1.5rc2 both treat range(0,1;3,4) simply as range(0,1;4).
Can treating the stream (3,4) as (4) be justified?

To sharpen the issue, compare:
```
jq -n 'range(0; (1,2))'
0
1

jq -n '(1,2) | range(0; . )'
0
0
1
```

**Comment by nicowilliams** [comment]
Yes, this is a bug in the bytecoded `range/2`.  The workaround is to use `range/3` with `1` as the third expression.  Do we need to fix this in 1.5?  I think no: it can be a known and documented bug to be fixed later, but we would have to fix the docs to say so.  Or we can fix the docs and say that `range/2` works as it in fact does work now -- it's worked that way for a long time, so why change it now?  I believe the docs bug would be my fault.

**Comment by dtolnay** [comment]
I think it is worth fixing, just not in 1.5.
